### PR TITLE
DEV-354 Feature Specs

### PR DESCRIPTION
**Pull request by moseshll**
Initial go at using ActionDispatch::SystemTestCase for system/feature testing
- Registration workflow.
- Approval request workflow.
- Use Capybara/Selenium with Docker. No luck with Cuprite.

### CODE REVIEW

**Comment by aelkiss** [comment]
It's a little bit surprising to me that the spec as-is wouldn't work with the default `:rack_test` capybara adapter (i.e. without the full browser & js stuff). Probably worth a little bit of discussion.

**Comment by moseshll** [comment]
Using `:rack_test` unfortunately makes it ~~impossible~~ tricky to get Select2 institution popup menus to do anything useful.

**Comment by moseshll** [comment]
I was unable to get Cuprite to work in our hybrid Docker/Capybara setup so I abandoned it for now. I wish the tests ran a bit faster.

**Review by aelkiss — APPROVED** [review]
The tests look great to me. Might be worth some additional comments explaining who would be doing each step for what purpose. I'll add a couple examples.

File: test/system/ht_registrations_test.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+class HTRegistrationsTest < ApplicationSystemTestCase
+  test "visit index" do
+    visit_with_login ht_registrations_url
+    assert_selector "h1", text: "Current Registrations"
+    assert_selector "a.btn", text: "New Registration"
+  end
+
+  test "registration workflow" do
+    visit_with_login ht_registrations_url
+    click_on "New Registration"
```
Comment (aelkiss) [inline test/system/ht_registrations_test.rb:12]:
```
# HathiTrust staff enters user registration details
```

File: test/system/ht_registrations_test.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    select "1 year", from: "Expire Type"
+    fill_in "Auth Rep Name", with: "Test Registration Auth Rep"
+    fill_in "Auth Rep E-mail", with: "test_reg_auth_rep@default.invalid"
+    fill_in "Auth Rep Date", with: "01/01/2022"
+    fill_in "Contact Info", with: "contact_info@default.invalid"
+    fill_in "HathiTrust Authorizer", with: "nobody@hathitrust.org"
+    # Capybara's too dumb to find and click a checkbox
+    check "ht_registration_mfa_addendum"
+    click_on "Submit Changes"
+    # Edit email page
```
Comment (aelkiss) [inline test/system/ht_registrations_test.rb:31]:
```
# HT staff reviews and edits email to be sent to new user
```

File: test/system/ht_registrations_test.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    assert_selector "h1", text: "Test Registration Applicant"
+    # Extract finalize URL from alert (only show in development/test environments)
+    assert_selector "div.alert-success"
+    link = first("div.alert-success").text.match(/http.+/)[0]
+    # Pretend we're the recipient of the registration e-mail and follow link.
+    visit link
+    click_on "Confirm Registration"
+    assert_content "confirmed for test_reg_applicant@default.invalid"
+    reg_id = HTRegistration.where(applicant_email: "test_reg_applicant@default.invalid").first.id
+    # Visit show page for new registration
```
Comment (aelkiss) [inline test/system/ht_registrations_test.rb:46]:
```
# HT staff confirm registration details and creates user
```

**Comment by aelkiss** [comment]
Rebasing to run feature specs including latest PR.

**Comment by aelkiss** [comment]
If at some point we have end-to-end authentication working then we should incorporate that into the tests, but this is certainly sufficient for now.

**Pull request merged**